src/blocks/file-blocks/explain/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { FileBlockProps, getLanguageFromFilename } from "@githubnext/utils";
+import { ThemeProvider } from "@primer/react";
+import { useCallback, useState } from "react";
+import { QueryClient, QueryClientProvider, useQuery } from "react-query";
+import { getHighlighter, setCDN } from "shiki";
+import { ExplanationComponent } from "./explanation";
+import "./index.css";
+import { Line } from "./line";
+import { LineMenu } from "./line-menu";
+import { LoadingState } from "./loading-state";
+
+setCDN("https://unpkg.com/shiki/");
+
+export interface Explanation {
+  code: string;
+  start: number;
+  end: number;
+}
+
+async function highlightCode(code: string, language: string) {
+  const highlighter = await getHighlighter({
+    theme: "github-light",
+    langs: ["js", "javascript"],
+  });
+  return highlighter.codeToThemedTokens(code, "js");
+}
+
+function BlockInner(props: FileBlockProps) {
+  const { context, content } = props;
+  const [localContent, setLocalContent] = useState(content);
```

Comment:
I don't see a use of the setter, could just pass `content` to `highlightCode` below

Reply:
Good catch, old code path which I will delete :+1: